fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt

```diff
@@ -0,0 +1,145 @@
+                        ),
+                    )
+                }
+
+                if (
+                    toSelected != null && fromSelected != null &&
+                    translationsDialogStore.state.translatedPageTitle == null
+                ) {
+                    translationsDialogStore.dispatch(
+                        TranslationsDialogAction.UpdateTranslatedPageTitle(
+                            context.getString(
+                                R.string.translations_bottom_sheet_title_translation_completed,
+                                fromSelected.localizedDisplayName,
+                                toSelected.localizedDisplayName,
+                            ),
+                        ),
+                    )
+                }
+
+                val translateFromLanguages = translationsState.supportedLanguages?.fromLanguages
+                translateFromLanguages?.let {
+                    translationsDialogStore.dispatch(
+                        TranslationsDialogAction.UpdateTranslateFromLanguages(
+                            translateFromLanguages,
+                        ),
+                    )
+                }
+
+                val translateToLanguages = translationsState.supportedLanguages?.toLanguages
+                translateToLanguages?.let {
+                    translationsDialogStore.dispatch(
+                        TranslationsDialogAction.UpdateTranslateToLanguages(
+                            translateToLanguages,
+                        ),
+                    )
+                }
+
+                if (translationsState.isTranslateProcessing) {
+                    updateStoreIfIsTranslateProcessing()
+                }
+
+                if (translationsState.isTranslated && !translationsState.isTranslateProcessing) {
+                    updateStoreIfTranslated()
+                }
+
+                if (translationsState.translationError != null) {
+                    translationsDialogStore.dispatch(
+                        TranslationsDialogAction.UpdateTranslationError(translationsState.translationError),
+                    )
+                }
+            }
+    }
+
+    private fun updateStoreIfIsTranslateProcessing() {
+        translationsDialogStore.dispatch(
+            TranslationsDialogAction.UpdateTranslationInProgress(
+                true,
+            ),
+        )
+        translationsDialogStore.dispatch(
+            TranslationsDialogAction.DismissDialog(
+                dismissDialogState = DismissDialogState.WaitingToBeDismissed,
+            ),
+        )
+    }
+
+    private fun updateStoreIfTranslated() {
+        translationsDialogStore.dispatch(
+            TranslationsDialogAction.UpdateTranslationInProgress(
+                false,
+            ),
+        )
+
+        translationsDialogStore.dispatch(
+            TranslationsDialogAction.UpdateTranslated(
+                true,
+            ),
+        )
```

Comment:
Perhaps the reducer `UpdateTranslated` should be setting the in-progress state to `false` when it is `true` to dispatching multiple actions to change a individual states?